projects/jdwp/codegen/dataclass_generator.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+
+from textwrap import dedent
+
+from projects.jdwp.defs.schema import (
+    Array,
+    ArrayLength,
+    Command,
+    IdType,
+    IntegralType,
+    Struct,
+    TaggedUnion,
+    UnionTag,
+)
+
+
+def format_enum_name(enum_value):
+    words = enum_value.name.split("_")
+    formatted_name = "".join(word.capitalize() for word in words)
+    return f"{formatted_name}Type"
+
+
+def map_id_type(id_type: IdType):
+    match id_type:
+        case IdType.OBJECT_ID:
+            return "ObjectIdType"
+        case IdType.THREAD_ID:
+            return "ThreadIdType"
+        case IdType.THREAD_GROUP_ID:
+            return "ThreadGroupIdType"
+        case IdType.STRING_ID:
+            return "StringIdType"
+        case IdType.CLASS_LOADER_ID:
+            return "ClassLoaderIdType"
+        case IdType.CLASS_OBJECT_ID:
+            return "ClassObjectIdType"
+        case IdType.ARRAY_ID:
+            return "ArrayIdType"
+        case IdType.REFERENCE_TYPE_ID:
+            return "ReferenceTypeIdType"
+        case IdType.CLASS_ID:
+            return "ClassIdType"
+        case IdType.INTERFACE_ID:
+            return "InterfaceIdType"
+        case IdType.ARRAY_TYPE_ID:
+            return "ArrayTypeIdType"
+        case IdType.METHOD_ID:
+            return "MethodIdType"
+        case IdType.FIELD_ID:
+            return "FieldIdType"
+        case IdType.FRAME_ID:
+            return "FrameIdType"
+        case _:
+            return f"Unrecognized type: {id_type}"
+
+
+def get_python_type_for_field(field_type, parent_struct_name="", field_name=""):
+    if isinstance(field_type, IdType):
+        return map_id_type(field_type)
+    elif (
+        isinstance(field_type, IntegralType)
+        or isinstance(field_type, ArrayLength)
+        or isinstance(field_type, UnionTag)
+    ):
+        return "int"
```

Comment:
We don't want to emit `ArrayLength` and `UnionTag` fields, we can compute appropriate values when serialising structs (for example by calling `len(arrayField)`)